spmd/tensor/parallel/api.py

```diff
@@ -86,3 +87,112 @@ def _shard_self_attn_params(name: str, module: nn.Module) -> None:
                 )
                 tp_multi_head_attention.copy(m)
                 module.register_module(n, tp_multi_head_attention)
+
+
+def _has_even_num_linears(module: nn.Module) -> bool:
+    """
+    We traverse through all the children of the given module and count the
+    number of Linear module. If the number is even, we return True.
+
+    Args:
+        module (nn.Module):
+            :class:``nn.Module`` object to be traversed and counted.
+
+    Return:
+        A boolean object which specifies whether the module contains
+        event-number of Linears in its children.
+
+    .. warning::
+        The traversal is not recursive for now.
```

Comment:
qq: why we need this function to be recursive?

Reply:
Because what if you have a sequence of a sequence of linear that kind of thing?